[CLI] Integrate `sui-test-validator` into `sui start` (#18204)
MystenLabs/sui: docs/content/guides/developer/getting-started/local-network.mdx

title: Connect to a Local Network
---

Use a Sui local network to test your dApps against the latest changes to Sui, and to prepare for the next Sui release to the Devnet or Testnet network. Sui CLI provides the `sui start` command to start a local network. There are several services that can be started when using `sui start`, such as an indexer, a faucet, or a local instance of the GraphQL service (including the online GraphiQL IDE). You can use the included faucet to get test SUI to use on the local network.

If you haven't already, you need to [install Sui CLI](./sui-install.mdx) on your system.

## Start the local network

To start the local network, run the following command after you installed Sui CLI.

```bash
RUST_LOG="off,sui_node=info" sui start --with-faucet --random-genesis
```

> This will:

Reviewer: ```suggestion
This command:
```
Nit: don't use future tense

Author: but but but, I like future tenses... me sad now.

Author: Addressed!

> * call the Sui CLI binary with two flags:

Reviewer: ```suggestion
* calls the Sui CLI binary with two flags:
```

Author: Addressed

 * `--with-faucet` to start a faucet service
 * `--random-genesis` to generate a new genesis and not persist the local network state

> * instruct Rust to set specific logging through the `RUST_LOG`=`off,sui_node=info` flags, which turns off logging for all components except `sui-node`. If you want to see more detailed logs, you can remove `RUST_LOG` from the command.

Reviewer: ```suggestion
* instructs Rust to set specific logging through the `RUST_LOG`=`off,sui_node=info` flags, which turns off logging for all components except `sui-node`. If you want to see more detailed logs, remove `RUST_LOG` from the command.
```

Author: Addressed

      --pg-port <PG_PORT>                      Port for the Indexer Postgres DB. Default port is 5432 [default: 5432]
      --pg-host <PG_HOST>                      Hostname for the Indexer Postgres DB. Default host is localhost [default: localhost]
      --pg-db-name <PG_DB_NAME>                DB name for the Indexer Postgres DB. Default DB name is sui_indexer [default: sui_indexer]
      --pg-user <PG_USER>                      DB username for the Indexer Postgres DB. Default username is postgres [default: postgres]
      --pg-password <PG_PASSWORD>              DB password for the Indexer Postgres DB. Default password is postgrespw [default: postgrespw]
      --epoch-duration-ms <EPOCH_DURATION_MS>  Set the epoch duration. Can only be used when `--random-genesis` flag is passed or if there's no genesis config and one will be auto-generated. When this flag is not set but
                                               `--random-genesis` is set, the epoch duration will be set to 60 seconds
      --no-full-node                           Start the network without a fullnode
```

Use `sui start --help` to see these options in your console.

### Persist local network state

> By default, when using `sui start` the command will use an existing Genesis and network configuration if the `~/.sui/sui_config` folder exists and if it has a `genesis.blob` file, or generate a new Genesis configuration if the folder does not exist. If the `--config-dir` is passed, it will check for the `genesis.blob` file and generate one if it does not exist.

Reviewer: ```suggestion
By default, when using `sui start` the command uses an existing genesis and network configuration if the `~/.sui/sui_config` folder exists and includes a `genesis.blob` file. If the folder doesn't exist, it creates the folder and generates a new genesis configuration. If you pass `--config-dir`, the command checks for the `genesis.blob` file and generates one if it does not exist.
```

Author: Addressed!

> Whenever you stop and start again the network without passing `--random-genesis` flag, all history will be preserved and will be accessible.

Reviewer: ```suggestion
Whenever you stop and start the network without passing the `--random-genesis` flag, all history is preserved and accessible.
```

Author: Addressed!
